01 SFC Hello VueJS/README.md

```diff
@@ -0,0 +1,323 @@
+    ]
+  }
+}
+
+```
+
+- Install `vue` library as project dependency:
+
+```
+npm install vue --save
+```
+
+- To use SFC there install vue-loader and vue-template-compiler:
+
+```
+npm install vue-loader vue-template-compiler --save-dev
+```
+
+- This library has its own TypeScript definition file, so we don't need install other typings.
+
+- Configure `webpack.config` to set `vue.js` as vendor:
+
+### ./webpack.config.js
+
+```diff
+...
+  entry: {
+    app: './main.ts',
++   vendor: [
++     'vue',
++   ],
+    vendorStyles: [
+      '../node_modules/bootstrap/dist/css/bootstrap.css',
+    ],
+  },
+...
+new ForkTsCheckerWebpackPlugin({
+      tsconfig: path.join(__dirname, './tsconfig.json'),
+      vue: true,
+    }),
+  ],
++ optimization: {
++   splitChunks: {
++     cacheGroups: {
++       vendor: {
++         test: /node_modules/,
++         name: 'vendor',
++         chunks: 'initial',
++         enforce: true
++       },
++     },
++   },
++ },  
+```
+
+- Update configuration for [`tsconfig.json`](https://vuejs.org/v2/guide/typescript.html#Recommended-Configuration):
+
+### ./tsconfig.json
+
+```diff
+{
+  "compilerOptions": {
+    "target": "es6",
+    "module": "es6",
+    "moduleResolution": "node",
+    "declaration": false,
+    "noImplicitAny": false,
+    "sourceMap": true,
+    "noLib": false,
+-   "suppressImplicitAnyIndexErrors": true
++   "suppressImplicitAnyIndexErrors": true,
++   "strict": true
+  },
+  "compileOnSave": false
+}
+
+```
+
+- There are [two builds available](https://vuejs.org/v2/guide/installation.html#Explanation-of-Different-Builds) to work with `Vue.js`.
+
+  - `Runtime + Compiler`: needed to compile templates on the client side, e.g. passing a string to `template` property.
+  - `Runtime-only`: when using a `pre-compiler` like `vue-loader`, `vueify`, etc.
+
+- Let's configure the use of `Runtime-only`. To configure it for webpack:
+
+### ./webpack.config.js
+
+```diff
+var path = require('path');
+var webpack = require('webpack');
+var HtmlWebpackPlugin = require('html-webpack-plugin');
+var MiniCssExtractPlugin = require('mini-css-extract-plugin');
+var ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
++ var VueLoaderPlugin = require('vue-loader/lib/plugin');
+...
+resolve: {
+-    extensions: ['.js', '.ts'],
++    extensions: ['.js', '.ts', '.vue'],
++    alias: {
++      'vue': 'vue/dist/vue.esm.js',
++    },
+  },
+...
+  module: {
+    rules: [
++     {
++       test: /\.vue$/,
++       exclude: /node_modules/,
++       loader: 'vue-loader',
++     },
+      {
+        test: /\.ts$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'ts-loader',
+          options: {
++           appendTsSuffixTo: [/\.vue$/],
+            transpileOnly: true,
+          },
+        },
+      },
+      {
+        test: /\.css$/,
+        use: [
+-         MiniCssExtractPlugin.loader,
++         process.env.NODE_ENV !== 'production'
++           ? 'vue-style-loader'
```

Comment:
Is this loader pending to install? In this readme I didn't see. I don't remember this code last PR review.

Reply:
vue-style-loader is a dependency of vue-loader